solutions/hw1/task1/matrices.py

```diff
@@ -14,26 +14,30 @@ def __init__(self, first: Vector, *args: Vector):
         self.content = [first, *args]
 
     def __eq__(self, other):
+        if not isinstance(other, Matrix):
+            raise NotImplementedError(
+                "An object of type matrix can only be compared with another object of type matrix"
+            )
         return self.content == other.content
 
     def __str__(self):
         return str(self.content)
 
-    def __add__(self, other):
+    def __add__(self, other: "Matrix") -> "Matrix":
         if self.get_dimensions() != other.get_dimensions():
             raise TypeError("Matrices must have the same dimensions")
         return Matrix(*[get_vector_sum(x, y) for x, y in zip(self.content, other.content)])
 
-    def get_dimensions(self):
+    def get_dimensions(self) -> dict:
```

Comment:
And a last change! `dict` is general. Use `Dict` from `typing` to make it exact. In this case it will be `Dict[str, int]`.